middleware/proxy/upstream.go

```diff
@@ -21,6 +22,14 @@ var (
 	supportedPolicies = make(map[string]func() Policy)
 )
 
+type upstreamEncoding uint
+
+const (
+	encodingNone upstreamEncoding = iota
+	encodingGRPC
```

Comment:
is grpc with tls an actual thing? Thought is actually required because http/2 sorta requires it. Can we just drop the non-tls part?